server/src/main/java/org/elasticsearch/action/search/SearchQueryThenFetchAsyncAction.java

```diff
@@ -72,4 +89,67 @@ protected void onShardGroupFailure(int shardIndex, Exception exc) {
     protected SearchPhase getNextPhase(final SearchPhaseResults<SearchPhaseResult> results, final SearchPhaseContext context) {
         return new FetchSearchPhase(results, searchPhaseController, context);
     }
+
+    ShardSearchRequest buildShardSearchRequestInternal(SearchShardIterator shardIt) {
+        ShardSearchRequest request = super.buildShardSearchRequest(shardIt);
+        if (resultConsumer == null) {
+            return request;
+        }
+        List<TopFieldDocs> topDocsList = resultConsumer.getRemainingTopDocs().stream()
+            .filter(obj -> obj instanceof TopFieldDocs)
+            .map(obj -> (TopFieldDocs) obj)
+            .collect(Collectors.toList());
+        if (topDocsList.isEmpty()) {
+            return request;
+        }
+        SearchSourceBuilder source = request.source();
+        int trackTotalHits = source.trackTotalHitsUpTo() == null ? DEFAULT_TRACK_TOTAL_HITS_UP_TO : source.trackTotalHitsUpTo();
+        long totalHits = -1;
+        FieldDoc bestBottom = null;
+        SortField primarySort = topDocsList.get(0).fields[0];
+        FieldComparator fieldComparator = primarySort.getComparator(1, 0);
+        int reverseMul = primarySort.getReverse() ? -1 : 1;
+        // we don't want to perform a costly merge to find the best bottom field doc so
+        // we just check pick the best bottom document among the available buffer. This
+        // means that we don't have the true-best bottom but this avoids running a partial
+        // merge too often.
+        for (TopFieldDocs topDocs : topDocsList) {
+            totalHits += topDocs.totalHits.value;
+            if (topDocs.scoreDocs.length == topDocsSize) {
+                FieldDoc cand = (FieldDoc) topDocs.scoreDocs[topDocsSize-1];
+                if (bestBottom == null ||
+                        fieldComparator.compareValues(cand.fields[0], cand.fields[0]) * reverseMul > 0) {
```

Comment:
should it be `fieldComparator.compareValues(bestBottom, cand.fields[0])....`

Reply:
Gloups, thanks. I renamed before pushing, this should be correct now.